.github/workflows/_helm-test.yaml

```diff
@@ -0,0 +1,190 @@
+####################################################################################################
+  helm-test:
+    runs-on: ${{ inputs.hardware }}
+    continue-on-error: true
+    outputs:
+      should_cleanup: ${{ steps.set_boolean.outputs.should_cleanup }}
+    steps:
+      - name: Clean Up Working Directory
+        run: sudo rm -rf ${{github.workspace}}/*
+
+      - name: Get checkout ref
+        id: get-checkout-ref
+        run: |
+          if [ "${{ github.event_name }}" == "pull_request" ] || [ "${{ github.event_name }}" == "pull_request_target" ]; then
+            CHECKOUT_REF=refs/pull/${{ github.event.number }}/merge
+          else
+            CHECKOUT_REF=${{ github.ref }}
+          fi
+          echo "CHECKOUT_REF=${CHECKOUT_REF}" >> $GITHUB_OUTPUT
+          echo "checkout ref ${CHECKOUT_REF}"
+
+      - name: Checkout out Repo
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ steps.get-checkout-ref.outputs.CHECKOUT_REF }}
+          fetch-depth: 0
+
+      - name: Set variables
+        env:
+          workload_path: ${{ inputs.workload }}
+        run: |
+          CHART_NAME=$(cat ${workload_path}/Chart.yaml |grep "^name" |cut -d ' ' -f 2)
+          USER_ID=$(whoami)
+          MODELPATH=/home/$USER_ID/.cache/huggingface/hub
+          echo "CHART_NAME=$CHART_NAME" >> $GITHUB_ENV
+          echo "MODELPATH=$MODELPATH" >> $GITHUB_ENV
+          echo "RELEASE_NAME=${CHART_NAME}$(date +%Y%m%d%H%M%S)" >> $GITHUB_ENV
+          echo "NAMESPACE=${CHART_NAME}-$(date +%Y%m%d%H%M%S)" >> $GITHUB_ENV
+          echo "ROLLOUT_TIMEOUT_SECONDS=600s" >> $GITHUB_ENV
+          echo "TEST_TIMEOUT_SECONDS=600s" >> $GITHUB_ENV
+          echo "KUBECTL_TIMEOUT_SECONDS=60s" >> $GITHUB_ENV
+          echo "should_cleanup=false" >> $GITHUB_ENV
+          echo "skip_validate=false" >> $GITHUB_ENV
+          echo "RELEASENAME=$RELEASE_NAME"
+          echo "NAMESPACE=$NAMESPACE"
+          echo "CHART_FOLDER=${workload_path}" >> $GITHUB_ENV
+
+      - name: helm lint test
+        run: |
+          helm repo add milvus https://zilliztech.github.io/milvus-helm
+          helm dependency update ${CHART_FOLDER}
+          if [[ ! $(helm lint ${CHART_FOLDER}) ]]; then
+            echo "Helm lint test failed"
+            exit 1
+          fi
+
+      - name: Initialize chart testing
+        run: |
+          # Replace values for CI test environment
+          HFTOKEN=${{ secrets.HFTOKEN }}
+          # HFTOKEN=$(cat /home/$USER_ID/.cache/huggingface/token)
+          if [ "${{ inputs.dockerhub }}" = "true" ]; then
+            echo "Use released docker images"
+          else
+            echo "Use internal docker registry"
+            # insert a prefix before opea/.*, the prefix is OPEA_IMAGE_REPO
+            # TBD, use global options.(https://github.com/opea-project/GenAIInfra/issues/562)
+            find . -name '*values.yaml' -type f -exec sed -i "s#repository: opea/*#repository: ${OPEA_IMAGE_REPO}opea/#g" {} \;
```

Comment:
if a PR changes the python code and helm chart simultaneously, our helm CI can't test the changes of the python code. Because the image won't be in internal OPEA_IMAGE_REPO until the PR is merged. We should follow the docker compose CI way to test this.

Reply:
Yes, this is an issue.
docker compose test will build all required images before deployment.
It run the build and deploy on the same host.
But for helm chart, the runner is dispatched to k8s master node (k8s-master), and the workload is actually running on workers(k8s-worker1). So we'll have to 1) leverage internal docker registry, or 2) modify the runner to k8s-worker1.
With 1), we'll need to setup steps mentioned above.
with 2), There should be one build.yaml for each component to build all required images for this comps (e.g. comps/embedding/docker_image_build/build.yaml). It can't be accomplished with this PR.